accounts/abi/bind/bindv2.go

```diff
@@ -0,0 +1,293 @@
+package bind
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"go/format"
+	"reflect"
+	"regexp"
+	"slices"
+	"strings"
+	"text/template"
+	"unicode"
+)
+
+// underlyingBindType returns the underlying Go type represented by the given type, panicking if it is not a pointer type.
+func underlyingBindType(typ abi.Type) string {
+	goType := typ.GetType()
+	if goType.Kind() != reflect.Pointer {
+		panic("trying to retrieve underlying bind type of non-pointer type.")
+	}
+	return goType.Elem().String()
+}
+
+// isPointerType returns true if the
+func isPointerType(typ abi.Type) bool {
+	return typ.GetType().Kind() == reflect.Pointer
+}
+
+type binder struct {
+	// contracts is the map of each individual contract requested binding
+	contracts map[string]*tmplContractV2
```

Comment:
What is this indexed by? Seems indexed by `type`, which seems to be the solidity-name of the contract?

Reply:
Yeah, the solidity name of the contract provided in the abi definition.

Reply:
All the string-keyed maps need docs, because sometimes the key is a library pattern, sometimes it is a unique id that we've created for a given struct.  It's not clear.